membership-attribute-service/app/controllers/StripeHookController.scala

```diff
@@ -1,35 +1,57 @@
 package controllers
 
+import com.gu.stripe.Stripe
 import com.typesafe.scalalogging.LazyLogging
 import play.api.libs.concurrent.Execution.Implicits._
 import com.gu.stripe.Stripe._
 import com.gu.stripe.Stripe.Deserializer._
-import components.{TestTouchpointComponents, NormalTouchpointComponents}
+import components.{NormalTouchpointComponents, TestTouchpointComponents}
+
 import scalaz.std.scalaFuture._
 import play.api.libs.json.Json
 import play.api.mvc._
+
 import scala.concurrent.Future
 import scalaz.OptionT
 
 
 class StripeHookController extends Controller with LazyLogging {
 
-  def process = Action.async { implicit request =>
+  def updatePrefs = Action.async { implicit request =>
     request.body.asJson.map(Json.fromJson[Event[StripeObject]](_)).fold[Future[Result]] {
       Future.successful(BadRequest("No JSON found in request body"))
     } { event =>
       (for {
         e <- OptionT(Future.successful(event.asOpt))
-        tp = if(e.liveMode) NormalTouchpointComponents else TestTouchpointComponents
+        tp = if (e.liveMode) NormalTouchpointComponents else TestTouchpointComponents
         eventFromStripe <- OptionT(tp.giraffeStripeService.Event.findCharge(e.id))
         identityId <- OptionT(tp.identityService.user(eventFromStripe.`object`.receipt_email))
         allowMarketing <- OptionT(Future.successful(eventFromStripe.`object`.metadata.get("marketing-opt-in").map(_ == "true")))
       } yield {
         tp.identityService.setMarketingPreference(identityId, allowMarketing)
         logger.info(s"$identityId marketing -> $allowMarketing")
       }).run
-      .map(_.fold[Result](Ok(Json.obj("updated" -> false)))
-                         (_ => Ok(Json.obj("updated" -> true))))
+        .map(_.fold[Result](Ok(Json.obj("updated" -> false)))
+          (_ => Ok(Json.obj("updated" -> true))))
+    }
+  }
+
+
+  def publishToSns = Action.async { implicit request =>
+    request.body.asJson.map(Json.fromJson[Event[StripeObject]](_)).fold[Future[Result]] {
+      Future.successful(BadRequest("No JSON found in request body"))
+    } { event =>
+      (for {
+        e <- OptionT(Future.successful(event.asOpt))
+        tp = if (e.liveMode) NormalTouchpointComponents else TestTouchpointComponents
```

Comment:
It feels odd to switch on the event, rather than have live events go to a different instance from test ones, but I don't know enough about Stripe to know whether this is a necessary evil.